src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```
@@ -65,135 +67,121 @@ public void UploadFile(string folderName, string fileName, Stream stream)
                     throw new ArgumentException($"This is not a valid certificate, file name: {_certificateFileName}", e);
                 }
             }
-                
-            request.UsePassive = true;
-            request.UseBinary = true;
-            request.KeepAlive = keepAlive;
 
-            return request;
+            client.Config.SocketKeepAlive = keepAlive;
+            if (_port != null)
+                client.Port = (int)_port;
+            else
+                client.Port = DefaultFtpPort;
+            client.Connect();
+            return client;
         }
 
         public void TestConnection()
         {
             if (_useSsl && string.IsNullOrWhiteSpace(_certificateAsBase64))
                 throw new ArgumentException("Certificate must be provided when using ftp with SSL!");
 
-            ExtractUrlAndDirectories(out var url, out _);
-            var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.ListDirectory, keepAlive: false);
-
-            try
+            using (var client = CreateFtpClient(_url, keepAlive: false))
             {
-                var response = request.GetResponse();
-                response.Close();
-            }
-            catch (WebException e)
-            {
-                var response = (FtpWebResponse)e.Response;
-                if (response.StatusCode == FtpStatusCode.ActionNotTakenFileUnavailable)
+                try
                 {
-                    // folder doesn't exist
-                    return;
+                    client.Connect();
+                }
+                catch (AuthenticationException e)
+                {
+                    throw new AuthenticationException("Make sure that you provided the correct certificate " +
+                                                      "and that the url matches the one that is defined in it", e);
+                }
+                catch (Exception e)
+                {
+                    throw new Exception();
```

lets throw the original exception here

---

Done